src/TesApi.Web/Events/RunnerEventsProcessor.cs

```diff
@@ -0,0 +1,516 @@
+                    .WithCancellation(cancellationToken))
+                {
+                    yield return uri.AbsoluteUri;
+
+                    switch (label)
+                    {
+                        case "stderr":
+                            stderr = stderr.Append(uri.AbsoluteUri);
+                            break;
+
+                        case "stdout":
+                            stdout = stdout.Append(uri.AbsoluteUri);
+                            break;
+                    }
+                }
+
+                if (alsoAddToExecutorLog)
+                {
+                    stderr = stderr.ToList();
+                    stdout = stdout.ToList();
+
+                    if (stderr.Any() || stdout.Any())
+                    {
+                        var log = tesTask.GetOrAddTesTaskLog().GetOrAddExecutorLog();
+
+                        if (stderr.Any())
+                        {
+                            log.Stderr = JsonArrayAsIndentedString(stderr);
+                        }
+
+                        if (stdout.Any())
+                        {
+                            log.Stdout = JsonArrayAsIndentedString(stdout);
+                        }
+                    }
+                }
+
+                static string JsonArrayAsIndentedString(IEnumerable<string> array)
+                    => System.Text.Json.JsonSerializer.Serialize(array, new System.Text.Json.JsonSerializerOptions(System.Text.Json.JsonSerializerOptions.Default) { WriteIndented = true });
+            }
+
+            async ValueTask AddProcessLogsAsync(Tes.Models.TesTask tesTask, CancellationToken cancellationToken)
+            {
+                await foreach (var _ in GetProcessLogsAsync(Tes.Runner.Events.EventsPublisher.ExecutorEndEvent, tesTask, cancellationToken).WithCancellation(cancellationToken))
+                { }
+            }
+        }
+
+        /// <summary>
+        /// Marks the event message as processed.
+        /// </summary>
+        /// <param name="message">Tes runner event message metadata.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> for controlling the lifetime of the asynchronous operation.</param>
+        /// <returns></returns>
+        /// <remarks>This method assumes <paramref name="message"/> was successfully validated by <see cref="ValidateMessageMetadata(RunnerEventsMessage)"/>.</remarks>
+        public async Task MarkMessageProcessedAsync(RunnerEventsMessage message, CancellationToken cancellationToken)
+        {
+            await azureProxy.SetBlobTags(
+                message.BlobUri,
+                message.Tags
+                    .Append(new(ProcessedTag, DateTime.UtcNow.ToString("O")))
+                    .ToDictionary(pair => pair.Key, pair => pair.Value),
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Prevents the message from being reprocessed by removing the 'task-id' tag. Used for malformed blobs.
```

Comment:
What is considered a malformed blob?

Reply:
`ArgumentException` (or derived) thrown from either `ValidateMessageMetadata`, `DownloadAndValidateMessageContentAsync`, or `GetMessageBatchStateAsync` (arguably only ValidateMessageMetadata() counts towards that definition.

Looking at the implementation, I think it would be better to either add the "processed" tag (or replace the last tag with one if there isn't room to add one) with a value like "error" or an empty string. I'll change that.